modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/CarouselView.tsx

```diff
@@ -3,14 +3,110 @@
  * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
  */
 
-import React from 'react';
+import {Card, ICardSchema} from '@liferay/frontend-data-set-web';
+import React, {useState} from 'react';
 
-const CarouselView = (props: any) => {
+import '../css/CarouselView.scss';
 
-	// eslint-disable-next-line no-console
-	console.log('props = ', props);
+const CarouselView = ({
+	items,
+	schema,
+}: {
+	items: Array<any>;
+	schema: ICardSchema;
+}) => {
+	const [selectedIndex, setSelectedIndex] = useState(0);
 
-	return <>Add real implementation here.</>;
+	const visibleItemsCount = 5;
+
+	const startIndex = Math.max(
+		0,
+		Math.min(
+			selectedIndex - Math.floor(visibleItemsCount / 2),
+			items.length - visibleItemsCount
+		)
+	);
+
+	const handleitemClick = (index: number) => {
+		setSelectedIndex(index);
+	};
+
+	const handlePrevClick = () => {
+		const newIndex =
+			selectedIndex === 0 ? items.length - 1 : selectedIndex - 1;
+		setSelectedIndex(newIndex);
+	};
+
+	const handleNextClick = () => {
+		const newIndex =
+			selectedIndex === items.length - 1 ? 0 : selectedIndex + 1;
+		setSelectedIndex(newIndex);
+	};
+
+	const {name, size, url} = items[selectedIndex];
```

Comment:
Just note that schema only provides mapping to data. URL must be in the data, and in [data this sample is using](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/java/com/liferay/frontend/data/set/sample/web/internal/portlet/FDSSamplePortlet.java#L107-L135) we currently don't have an image url field.

Not related to task, but @dsanz , at some point we should move this sample data definition and generation to batch engine :slightly_smiling_face:

Reply:
As far as I'm concerned, we can push this without the image, I'd just remove the broken bit from UI

![image](https://github.com/user-attachments/assets/3b86790f-74cb-4267-96a9-c8806f36ed73)

If we do want the image, it needs to be added to object definition and generation, as well as mapped in schema.

Reply:
> @chestofwonders now we have an schema, so we can get the data from the item using the props values indicated in the schema.
> 
> As for the image we'll need to add a new string field to the FDSSample object definition, which can be mapped in the schema.
> 
> Values for that field can just be the same image URL, such as `/image/company_logo` which is going to work though introduces an external dependency. I'm not worried because sample is meant to illustrate, we don't need to test this component thoroughly

@dsanz getting props values through schema were added [here](https://github.com/liferay-frontend/liferay-portal/pull/4576/commits/f473655feaa12343d3a854eb8204eec0ddc68299#diff-e2e4181e72a81b485b48ea55d2b37da52b400d6f277df555251d815c7debd78b). Did you miss something else related to that?

Reply:
Oh great then! I must have checked an older commit, sorry about re-asking.

> Not related to task, but @dsanz , at some point we should move this sample data definition and generation to batch engine 🙂

yes, that's pending. Perhaps we can wait till some test involving the sample fails

And finally, yep, image shall be in the data and populated. I suggested using a fixed value for it. This is what I had in mind:
![image](https://github.com/user-attachments/assets/e46bb7b0-6108-4cb8-aef0-6a373c7c5c2e)
(plus the mapping)

So, basically we can push like it is (because it's enough), or just add the image as follows (because it's cheap). 

I think this is opt-in so I'll let @chestofwonders decide to add or jsut forward as it is

Reply:
@dsanz I think it's a good improvement, I'm rather in favor of adding it